### PR TITLE
Optimize the comparison mode of bool judgment

### PR DESCRIPTION
**Pull request by Fish-pro**
Signed-off-by: Fish-pro <zechun.chen@daocloud.io>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Optimize the comparison mode of bool judgment

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @Fish-pro, what's the reason for your modification? I did a simple test, and it didn't seem to improve the efficiency.

image

**Comment by Fish-pro** [comment]
> Hi @Fish-pro, what's the reason for your modification? I did a simple test, and it didn't seem to improve the efficiency.
> 
> image

Hi @XiShanYongYe-Chang 
There are two reasons for the change:
1. I have seen that the judgments of Boolean values are various in the code, such as `tt.expectError! = true` and `tt.expectError == false`, which are poor for code readability
2. I have to focus on the like [cilium](https://github.com/cilium/cilium), in the pr have made the same uniform, see https://github.com/cilium/cilium/pull/22588/files in detail

I ran tests locally, too, and they're the same as yours
```sh
goos: darwin
goarch: amd64
pkg: github.com/Fish-pro/pprof-demo/pkg/teststudy/iftest
cpu: Intel(R) Core(TM) i5-6360U CPU @ 2.00GHz
BenchmarkSumA-4         1000000000               3.172 ns/op
BenchmarkSumB-4         1000000000               3.146 ns/op
PASS
ok      github.com/Fish-pro/pprof-demo/pkg/teststudy/iftest     7.091s
```

**Comment by Fish-pro** [comment]
@XiShanYongYe-Chang 
Therefore, can we unify the places that do not conform to the norms as `==`, what's your opinion?

**Comment by XiShanYongYe-Chang** [comment]
In fact, I think it's probably just a matter of coding habits.

**Comment by XiShanYongYe-Chang** [comment]
Ask for some advice from @RainbowMango @iawia002 @Poor12

**Comment by RainbowMango** [comment]
Maybe we need more evidence to do this. For now, we don't have such programming specifications.

**Comment by Fish-pro** [comment]
> Maybe we need more evidence to do this. For now, we don't have such programming specifications.

@RainbowMango @XiShanYongYe-Chang 
Thanks for tips， I see what you mean，now I have changed it to simple unification, please evaluate whether it is meaningful. If it is not needed for the time being, please remind me to close this pr, thanks.

File: pkg/util/helper/patch_test.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func TestGenMergePatch(t *testing.T) {
 			patch, err := GenMergePatch(testObj, tc.modifyFunc())
 			if err != nil && tc.expectErr == false {
 				t.Fatalf("unexpect error, but got: %v", err)
-			} else if err == nil && tc.expectErr == true {
+			} else if err == nil && tc.expectErr {
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/patch_test.go:91]: This modification is unnecessary.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Other parts lgtm

File: pkg/resourceinterpreter/defaultinterpreter/revisereplica_test.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func TestReviseDeploymentReplica(t *testing.T) {
 			if err == nil && tt.expectError == true {
 				t.Fatal("expect an error but got none")
 			}
-			if err != nil && tt.expectError != true {
+			if err != nil && tt.expectError == false {
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/revisereplica_test.go:74]: I still can't figure out the benefits of this change.
There is no difference between the two styles.

Comment (Fish-pro) [reply pkg/resourceinterpreter/defaultinterpreter/revisereplica_test.go:74]: @RainbowMango Ok, thanks. I'll close this pr

File: pkg/util/helper/patch_test.go (hunk 1 above)
Comment (Fish-pro) [reply pkg/util/helper/patch_test.go:91]: Ok, thanks. I'll close this pr

Comment (XiShanYongYe-Chang) [reply pkg/util/helper/patch_test.go:91]: In any case, still thank you for your contribution!

**Pull request closed without merging**